crates/uv-resolver/src/lock.rs

```diff
@@ -419,6 +455,19 @@ impl Lock {
             doc.insert("requires-python", value(requires_python.to_string()));
         }
 
+        // Write the settings that were used to generate the resolution.
+        // This enables us to invalidate the lockfile if the user changes
+        // their settings.
+        if self.resolution_mode != ResolutionMode::default() {
```

Comment:
I decided to omit these if users use the default. It means we may have the "wrong" behavior if we change the default, but I think that's pretty unlikely, and even still, it's kind of fine -- we'd just regenerate their lock. The upside is that we almost never write these, so the lockfile remains succinct.